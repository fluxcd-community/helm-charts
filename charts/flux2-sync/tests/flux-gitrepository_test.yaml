suite: test flux-gitrepository
templates:
  - flux-gitrepository.yaml
tests:
  - it: should match snapshot of default values
    capabilities:
      majorVersion: 1
      minorVersion: 19
    set:
      gitRepository:
        spec:
          url: repository_url
    asserts:
      - matchSnapshot: {}
  - it: should match snapshot with special values
    capabilities:
      majorVersion: 1
      minorVersion: 19
    set:
      gitRepository:
        spec:
          url: repository_url
          ref: {"branch": "branch"}
          ignore: |
            # exclude all
            /*
            # include deploy dir
            !/cluster-name/fluxcd-applications
    asserts:
      - matchSnapshot: {}
