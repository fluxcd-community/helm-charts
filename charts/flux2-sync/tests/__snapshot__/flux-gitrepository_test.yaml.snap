should match snapshot of default values:
  1: |
    apiVersion: source.toolkit.fluxcd.io/v1
    kind: GitRepository
    metadata:
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        helm.sh/chart: flux2-sync-1.10.0
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      interval: 5m
      ref:
        branch: master
      url: repository_url
should match snapshot with special values:
  1: |
    apiVersion: source.toolkit.fluxcd.io/v1
    kind: GitRepository
    metadata:
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        helm.sh/chart: flux2-sync-1.10.0
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ignore: |
        # exclude all
        /*
        # include deploy dir
        !/cluster-name/fluxcd-applications
      interval: 5m
      ref:
        branch: branch
      url: repository_url
